fastlane/Fastfile

```diff
@@ -1,78 +1,51 @@
 default_platform(:ios)
 
 platform :ios do
-  before_all do
-    cocoapods
-  end
-
-  desc 'Run tests on the Core framework of SDK'
+  desc 'Run tests on all targets'
   lane :test_all do
     test_core
-    test_publisher
+    test_internal
     test_subscriber
+    test_publisher
   end
 
-  desc 'Run tests on the Core framework of SDK'
+  desc 'Run Core tests'
   lane :test_core do
-    scan(
-      workspace: './AblyAssetTracking.xcworkspace',
-      scheme: 'Core',
-      clean: true,
-      code_coverage: true,
-    )
-
-    # Clean previous test coverage and generate new one
-    sh('rm -rf ./coverage_core')
-    slather(
-      workspace: './AblyAssetTracking.xcworkspace',
-      proj: 'Core/Core.xcodeproj',
-      scheme: 'Core',
-      output_directory: 'coverage_core',
-      html: true,
-      ignore: ['Core/Tests/*'],
-      source_directory: './Core/Sources'
+    xcodebuild(
+      scheme: 'CoreTests',
+      configuration: 'Debug',
+      destination: 'platform=iOS Simulator,name=iPhone 12',
+      test: "test"
     )
   end
 
-  desc 'Run tests on the Publisher framework of SDK'
-  lane :test_publisher do
-    scan(
-      workspace: './AblyAssetTracking.xcworkspace',
-      scheme: 'Publisher',
-      clean: true,
-      code_coverage: true,
-    )
-
-    # Clean previous test coverage and generate new one
-    sh('rm -rf ./coverage_publisher')
-    slather(
-      workspace: './AblyAssetTracking.xcworkspace',
-      proj: 'Publisher/Publisher.xcodeproj',
-      scheme: 'Publisher',
-      output_directory: 'coverage_publisher',
-      html: true,
-      ignore: ['Core/*', 'Common/*', 'Publisher/Tests/*'],
-      source_directory: './Publisher/Sources'
+  desc 'Run Internal tests'
+  lane :test_internal do
+    xcodebuild(
+      scheme: 'InternalTests',
+      configuration: 'Debug',
+      destination: 'platform=iOS Simulator,name=iPhone 12',
+      test: "test"
     )
   end
 
-  desc 'Run tests on the Subscriber framework of SDK'
+  desc 'Run Subscriber tests'
   lane :test_subscriber do
-    scan(
-      workspace: './AblyAssetTracking.xcworkspace',
-      scheme: 'Subscriber',
-      clean: true,
-      code_coverage: true,
+    xcodebuild(
+      scheme: 'SubscriberTests',
+      configuration: 'Debug',
+      destination: 'platform=iOS Simulator,name=iPhone 12',
+      test: "test"
```

Comment:
>Note: xcodebuild is a complex command, so it is recommended to use gym for building your ipa file and scan for testing your app instead.
https://docs.fastlane.tools/actions/xcodebuild/

`Scan` won't work?

Reply:
Did you try this solution? https://uptech.team/blog/swift-package-manager 
**chapter**: `2. To Change The Path Of SPM Dependencies, You Need To Override ClonedSourcePackagesDirPath.`

Reply:
> EDIT: I guess this is the reason

Yes, fastlane scan does not support Swift Packages.

> Did you try this solution?

I'm not sure what you mean. How is changing SPM dependency path a solution? Do you suggest I cache the SPM packages to improve build times? I don't think scan does that? I think that's could be a good github issue to speed up dependency installation, especially since Mapbox dependencies could reach ~100MB.

Reply:
I mean if `scan` some support for SPM (in this case `cloned_source_packages_path`) I guess it should support SPM somehow?